### PR TITLE
Simplify "slice(concat(x, y,...))" to x if applicable

### LINKED ISSUE #2734: Simplify "slice(concat(x))" to  just "x" where applicable
### Problem Description

This sample is from SD-XL UNet model 

```
@24 = gpu::code_object[code_object=9344,symbol_name=sin_kernel,global=160,local=1024,](@22,@23) -> half_type, {2, 160}, {160, 1}, target_id=0: 0.0177847ms, 1%

@26 = gpu::code_object[code_object=9344,symbol_name=cos_kernel,global=160,local=1024,](@22,@25) -> half_type, {2, 160}, {160, 1}, target_id=0: 0.0182076ms, 1%

@28 = gpu::code_object[code_object=9160,symbol_name=concat_kernel,global=320,local=1024,](@24,@26,@27) -> half_type, {2, 320}, {320, 1}
@29 = slice[axes={1},starts={0},ends={160}](@28) -> half_type, {2, 160}, {320, 1}
@30 = load[offset=11264,end=12544](@1) -> half_type, {2, 320}, {320, 1}
@31 = slice[axes={1},starts={160},ends={320}](@28) -> half_type, {2, 160}, {320, 1}
@32 = gpu::code_object[code_object=9160,symbol_name=concat_kernel,global=320,local=1024,](@31,@29,@30) -> half_type, {2, 320}, {320, 1}
```

Notice that first concat takes @24 as the first arg and then @29 is slicing the same. Thereforre @29 can be just replaced with @24. 

Similarly for the slice @31 it can be replaced with @26. 

### Operating System

22.04

### CPU

Any

### GPU

AMD Instinct MI300

### Other

_No response_

### ROCm Version

ROCm 6.0.0

### Steps to Reproduce

_No response_

### (Optional for Linux users) Output of /opt/rocm/bin/rocminfo --support

_No response_

### Additional Information

_No response_

### PR DESCRIPTION
**Pull request by umangyadav**
Fixes #2734

### CODE REVIEW

File: test/simplify_reshapes_test.cpp [hunk 1]
```diff
@@ -556,6 +556,252 @@ TEST_CASE(nested_squeeze_reshape)
...
+    }
+    migraphx::module m2 = m1;
+    run_pass(m1);
+    EXPECT(m1 == m2);
+}
+
+TEST_CASE(concat_slice_multiple_slice_use)
+{
+    // multiple use for slice1 and slice3 but single use for slice2
+    // slice2 is replaced with original input to concat
```
[the remark this answers is not in the document]
Comment (umangyadav) [reply test/simplify_reshapes_test.cpp:725]: Yes we can eliminate all slices. Let me change this. i had "used_once()" as one of the assumptions like many other matchers, but doesn't apply in this case.

[the remark this answers is not in the document]
Comment (umangyadav) [reply test/simplify_reshapes_test.cpp]: Pushed changes.

**Review by shivadbhavsar — APPROVED** [review]

**Comment by pfultz2** [comment]
Can you add a test case where one slice can be replaced and the other cannot because it cross the boundaries of the other inputs?

**Comment by umangyadav** [comment]
> Can you add a test case where one slice can be replaced and the other cannot because it cross the boundaries of the other inputs?

Added.

**Review by pfultz2 — APPROVED** [review]
